src/PHPStan/Reflection/TypeToCallReflectionResolver/TypeToCallReflectionResolverInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\PHPStan\Reflection\TypeToCallReflectionResolver;
+
+use PHPStan\Reflection\ClassMemberAccessAnswerer;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\Type;
+
+interface TypeToCallReflectionResolverInterface
+{
+    public function supports(Type $type): bool;
+
+    /**
+     * @return FunctionReflection|MethodReflection|null
+     */
+    public function resolve(Type $type, ClassMemberAccessAnswerer $classMemberAccessAnswerer);
```

Comment:
```suggestion
    public function resolve(Type $type, ClassMemberAccessAnswerer $classMemberAccessAnswerer): ?\ReflectionFunctionAbstract`
```

Reply:
Won't work. We're not returning a core reflection instance, but an instance of PHPStan's reflection (or null). A return type is not possible here.

Reply:
What about object?

Reply:
Do they have any common parent?

Reply:
I checked and no. FunctionReflection and MethodReflection are two independant interfaces.

Reply:
:(